### PR TITLE
[WEB-3355] housekeeping

### PR DESCRIPTION
**Pull request by tadam313**
* integrate `karma` + `jasmine` running on `JSDOM` to do the tests
* using `nvm` to keep node version in sync with CI
* load angular from npm instead of local dep
* added `test-client` bitrise step
* fix existing unit tests
* renamed tests to `xs.spec.js`
* updated README

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+lts/Dubnium
```
Comment (tadam313) [inline .nvmrc]: CI is using v10. node as that's the latest official for ubuntu

File: Gemfile [hunk 2]
```diff
@@ -4,9 +4,6 @@ gem "nokogiri", '~> 1.10.4'
 gem "middleman", '~> 3.4.1'
 gem "middleman-livereload"
 gem "slim", '~> 3'
-gem "underscore-rails"
-gem "jquery-middleman"
-gem "middleman-jasmine"
```
Comment (tadam313) [inline Gemfile]: do not need these anymore, coming from npm

File: config.rb [hunk 3]
```diff
@@ -62,11 +60,6 @@ def svg(filename)
...
-sprockets.import_asset "underscore"
-sprockets.import_asset "angular"
-sprockets.import_asset "angular-route"
-sprockets.import_asset "angular-sanitize"
-sprockets.import_asset "angular-animate"
-sprockets.import_asset "angular-cookies"
-sprockets.import_asset "angular-mocks"
+# in order to support require sprockets annotations
+sprockets.append_path File.join root, 'node_modules'
+sprockets.append_path File.join root, 'vendor-js'
```
Comment (tadam313) [inline config.rb]: added vendor-js and node modules include path

File: karma.conf.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+process.on('infrastructure_error', (error) => {
+  console.error('infrastructure_error', error);
+});
+
+
+module.exports = (config) => {
+  config.set({
+
+    // base path that will be used to resolve all patterns (eg. files, exclude)
+    basePath: 'build',
```
Comment (tadam313) [inline karma.conf.js]: using the build JS bundle by middleman

File: spec/javascripts/step.spec.js [hunk 5]
```diff
@@ -7,17 +7,17 @@ describe("Step", function() {
 		Step = _Step_;
 	}));
 
-	describe("title", function() {
```
Comment (erosdome) [inline spec/javascripts/step.spec.js]: Naming the file with capital, *Step.spec.js* instead of *step.spec.js*? To match the class's name.

Comment (tadam313) [reply spec/javascripts/step.spec.js]: Fair enough, will do it

File: package.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/bitrise-io/bitrise-workflow-editor.git"
+  },
+  "keywords": [
+    "workflow",
+    "steps"
+  ],
+  "author": "",
+  "license": "ISC",
```
[the remark this answers is not in the document]
Comment (tadam313) [reply package.json]: will change it

**Pull request merged**